### PR TITLE
study view: Add localdb tests for x vs y charts

### PR DESCRIPTION
**Pull request by adamabeshouse**
Also fix some UI issues

### CODE REVIEW

File: src/pages/studyView/chartHeader/ChartHeader.tsx [hunk 1]
```diff
@@ -689,7 +689,7 @@ export class ChartHeader extends React.Component<IChartHeaderProps, {}> {
                             >
                                 <button
                                     className={classnames(
-                                        'btn btn-xs btn-default',
+                                        'btn btn-xs btn-default deleteChart',
```
Comment (dippindots) [inline src/pages/studyView/chartHeader/ChartHeader.tsx:692]: If this is only for testing purpose, maybe we can use `data-test={'deleteChart'}`?

File: end-to-end-test/local/specs/core/studyview.screenshot.spec.js [hunk 2]
```diff
@@ -54,3 +56,87 @@ describe('study view generic assay categorical/binary features', function() {
...
+            timeout: WAIT_FOR_VISIBLE_TIMEOUT,
+        });
+        $(ADD_CHART_X_VS_Y_TAB).click();
+
+        // Add Mutation Count vs Diagnosis Age chart
+        // X-axis: diagnosis age
+        $('.xvsy-x-axis-selector').waitForDisplayed();
+        $('.xvsy-x-axis-selector').click();
+        $('.xvsy-x-axis-selector #react-select-2-option-0').waitForDisplayed();
+        $('.xvsy-x-axis-selector #react-select-2-option-0').click();
```
Comment (dippindots) [inline end-to-end-test/local/specs/core/studyview.screenshot.spec.js:102]: One question about this `#react-select-2-option-0`, will this always work? I am not so sure about how react-select assign id to components...

**Review by dippindots — APPROVED** [review]
Looks good to me, just two minor comments.

File: end-to-end-test/local/specs/core/studyview.screenshot.spec.js (hunk 2 above)
Comment (adamabeshouse) [reply end-to-end-test/local/specs/core/studyview.screenshot.spec.js:102]: it's definitely worrisome... I will look into if I can query the options by text content

**Review by dippindots — APPROVED** [review]
Looks good to me, thanks!

**Review by alisman — APPROVED** [review]

**Pull request merged**